lib/core/test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -434,6 +436,34 @@ expectPathEventuallyExist filepath = do
             False ->
                 threadDelay (oneSecond `div` 2) >> doesPathExistNow
 
+-- | A file is eventually created on the given location
+expectProcStdOutHas :: CreateProcess -> Text -> IO ()
+expectProcStdOutHas procc out = do
+    handle <- async (runProcUntil procc out)
+    winner <- race (threadDelay (60 * oneSecond)) (wait handle)
+    case winner of
+        Left _ -> expectationFailure $
+            "waited more than 60s for proccess to have in stdout = " ++ T.unpack out
+        Right _ ->
+            return ()
+  where
+    oneSecond = 1_000_000
+
+    runProcUntil :: CreateProcess -> Text -> IO ()
+    runProcUntil pr wants = withCreateProcess pr $ \_ (Just stdout) _ h -> do
+           hSetBuffering stdout LineBuffering
+           waitForIt stdout h
+     where
+         waitForIt stdout h = do
+             is <- TIO.hGetLine stdout
```

Comment:
I don't think GHC runtime will be happy if we try to `hGetLine` on an empty handle. There's maybe some wild IO exception error handling to do here :s

Reply:
I've added some handling there...